### PR TITLE
184084119 roster upload

### PR DESCRIPTION
**Pull request by Rohithone**
# PR Checklist

Please check if your PR fulfills the following requirements
- [ ] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

# What is the ticket # detailing the issue?

Ticket:  https://www.pivotaltracker.com/n/projects/2481183/stories/184084119

# A brief description of the changes

Current behavior: Timeout Error When Using ER Roster Upload Tool for Senate

New behavior: Added event source to avoid timeout error

# Environment Variable

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. Please share the name of the variable below that would enable/disable the feature and which client it applies to.

Variable name: PROCESS_EVENT_SOURCE_IS_ENABLED'

- [x] DC
- [ ] ME

### CODE REVIEW

**Review by polographer — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ gem 'database_cleaner',       '~> 1.7'
 # Local components/engines
 #######################################################
 gem 'acapi',              git: "https://github.com/ideacrew/acapi.git", branch: 'amqp_proc_title'
-gem 'aca_entities',       git: 'https://github.com/ideacrew/aca_entities.git', branch: 'trunk'
+gem 'aca_entities',       git: 'https://github.com/ideacrew/aca_entities.git', branch: '184084119-roster-upload-channels'
```
[the remark this answers is not in the document]
Comment (Rohithone) [reply Gemfile]: I changed to trunk can you please merge aca_entities PR

File: Gemfile.lock [hunk 2]
```diff
@@ -7,8 +7,8 @@ GIT
 
 GIT
   remote: https://github.com/ideacrew/aca_entities.git
-  revision: a6ff3970906729e388c1b1dfcaa7b849361f12b8
-  branch: trunk
+  revision: f9fc69d75156c76ccc93f704947ffeffe7a307e2
+  branch: 184084119-roster-upload-channels
```
[the remark this answers is not in the document]
Comment (Rohithone) [reply Gemfile.lock]: I changed to trunk can you please merge aca_entities PR

File: components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/employers/employer_profiles_controller.rb [hunk 3]
```diff
@@ -130,19 +131,14 @@ def bulk_employee_upload
...
-                flash[:notice] = "#{roaster_upload_count} records uploaded from CSV"
-                redirect_to URI.parse(@roster_upload_form.redirection_url).to_s
-              else
-                render @roster_upload_form.redirection_url || default_url
-              end
-            rescue StandardError => e
-              @roster_upload_form.errors.add(:base, e.message)
-              render @roster_upload_form.redirection_url || default_url
-            end
+            file_name = file.original_filename
```
[the remark this answers is not in the document]
Comment (Rohithone) [reply components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/employers/employer_profiles_controller.rb]: I added a feature flag, please verify

File: app/event_source/subscribers/file_upload/roster_parser_requested_subscriber.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Subscribers
+  module FileUpload
+  # Subscriber will receive Enterprise requests like date change
+    class RosterParserRequestedSubscriber
```
[the remark this answers is not in the document]
Comment (Rohithone) [reply app/event_source/subscribers/file_upload/roster_parser_requested_subscriber.rb:6]: Updated code, please check

**Pull request closed without merging**